pythonlib/src/mill/pythonlib/SetupTools.scala

```diff
@@ -0,0 +1,56 @@
+package mill.pythonlib
+
+import mill._
+
+/**
+ * This trait exists solely to distinguish tasks that wrap python's setuptools
+ * to build distributions, from other, more common tasks that are defined in
+ * `PythonModule`.
+ *
+ * Essentially, every `PythonModule` should have a way to build wheels and
+ * sdists, however how that is done can vary. As of now, we only support
+ * building artifacts via setuptools (generating and invoking a `setup.py`), but
+ * in the future that may change. For example, we may support building wheels by
+ * reading PEP-517 compliant `pyproject.toml` files, or even use "pure" Mill to
+ * build wheels and sdists. Hence, to separate concerns, we define some tasks in
+ * this mixin, and extend it in PythonModule.
+ */
+trait SetupTools extends Module { self: PythonModule =>
```

Comment:
not sure about this design

Reply:
Or, we KISS it, and flatten everything into one trait. Extracting the tasks will then break binary compatibility, but I'm sure that will happen at least once anyway, given that `pythonlib` is brand new anyway,

Reply:
I'd say dont worry about bincompat or stability at all for now; its still to early in the lifecycle of this part of the project. I'd just flatten it for now until we have a concrete reason to abstract it and then make the change